core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -439,45 +439,21 @@ public Object readGenericValue() throws IOException {
             case 5:
                 return readBoolean();
             case 6:
-                int bytesSize = readVInt();
-                byte[] value = new byte[bytesSize];
-                readBytes(value, 0, bytesSize);
-                return value;
+                return fastReadByteArray();
```

Comment:
@jpountz Because there is already a method named `readByteArray` and that method reads a single byte at a time but the method here uses `readBytes` which can be optimized to read a chunk of bytes. I *think* that these methods can be combined but will address that in a follow-up if so to keep this PR purely about inlining.